Makefile.package

```diff
@@ -93,16 +93,20 @@ endif
 # These options can be set to influence package and configure.
 CONAN_OPTIONS :=
 
-.PHONY: ${BUILD_LOCKFILE}
 ifneq "${LOCKFILE_SOURCE}" ""
+.PHONY: ${BUILD_LOCKFILE}
 ifeq "$(realpath ${LOCKFILE_SOURCE})" "$(realpath ${SOURCE_CONANFILE})"
 $(error "LOCKFILE_SOURCE must contain superset of SOURCE_CONANFILE package and dependencies")
 endif
 LOCKFILE_OPTION := --lockfile="${BUILD_LOCKFILE}"
-${BUILD_LOCKFILE}: ${LOCKFILE_SOURCE} | ${BUILD_DIR} export
+${BUILD_LOCKFILE}: ${LOCKFILE_SOURCE} export
```

Comment:
I wonder about the utility of the `export` prerequisite. What we need is that all requirements in `LOCKFILE_SOURCE` exist in the conan cache. For the (common) use case of an in-source build using the repository root `conanfile.py`, we would need to do `make export` in the repository root. However, we probably do not want to do this for every build and I cannot think about a general solution in this moment.

Reply:
So here's my reasoning:

Normally, when doing any other things that create the package, an export is implicit:

- `make all` uses the sources as-is
- `make package` performs an export of the package itself first

So you would expect something like the following reflexive to be inherently valid and just work:

    make LOCKFILE_SOURCE=conanfile.py package

But it doesn't! Because lockfile generation happens before the package creation.

Now, most often we'll just get "cannot find package ZZZ version X.Y" error messages. But you won't if you for example make some changes in some conanfile.py and don't export it first, which is unorthogonal to how the rest of the makefiles work.

Without the export, the following would lead to errors:

    cd engine              # Currently working on engine
    ${EDITOR} conanfile.py # Make some changes to dependencies
    make package LOCKFILE_SOURCE=../conanfile.py
    
This is for me, a common use case, and one which the `export` prerequisite solves.

Reply:
Ok, I haven't thought about the package target - makes sense. Thanks for the detailed explanation!